File: app-server-webflux/src/main/java/com/github/arhor/simple/expense/tracker/service/mapper/ExpenseItemMapper.java
package com.github.arhor.simple.expense.tracker.service.mapper;

import org.mapstruct.InheritInverseConfiguration;
import org.mapstruct.Mapper;
import org.mapstruct.Mapping;

import com.github.arhor.simple.expense.tracker.data.model.ExpenseItem;
import com.github.arhor.simple.expense.tracker.model.ExpenseItemDTO;

@Mapper(config = SharedMappingConfig.class)
public interface ExpenseItemMapper {

    @Mapping(target = "id", ignore = true)
    ExpenseItem mapToEntity(ExpenseItemDTO dto, Long expenseId);

    @InheritInverseConfiguration
    ExpenseItemDTO mapToDTO(ExpenseItem entity);
}

File: apps/server/src/main/kotlin/com/github/arhor/simple/expense/tracker/service/mapping/ExpenseItemMapper.kt
package com.github.arhor.simple.expense.tracker.service.mapping

import com.github.arhor.simple.expense.tracker.data.model.ExpenseItem
import com.github.arhor.simple.expense.tracker.model.ExpenseItemRequestDTO
import com.github.arhor.simple.expense.tracker.model.ExpenseItemResponseDTO

interface ExpenseItemMapper {

    fun mapToEntity(dto: ExpenseItemRequestDTO, expenseId: Long): ExpenseItem

    fun mapToDTO(entity: ExpenseItem): ExpenseItemResponseDTO
}
